docs/proposals/central-segmented-registry.md

```diff
@@ -0,0 +1,144 @@
+
+        $ oc new-project test
+        $ oadm registry --namespace test
+        $ oc expose service docker-registry --namespace test
+
+The exposed route provides a unique endpoint for referencing the registry.
+
+Each registry has its own storage backend.
+
+### Managing the Clusters
+
+The central registry is referenced using external service abstraction. Instead of deploying a registry on each cluster, Service and Route objects are created in the "default" project namespace on each cluster.
+
+        - kind: Service
+          apiVersion v1
+          metadata:
+            name: docker-registry
+          spec:
+            ports:
+              - name: docker-registry
+                protocol: TCP
+                port: 443
+                targetPort: 443
+                nodePort: 0
+          selector: {}
+        - kind: Route
+          apiVersion: v1
+          metadata:
+            labels:
+              app: docker-registry
+            name: docker-registry
+          spec:
+            tls:
+              ...
+            to:
+              kind: Service
+              name: docker-registry
+          status: {}
+
+An Endpoint object created in the "default" project namespace on each cluster. In this example, the test OpenShift cluster references the test registry instance in the central registry cluster.
+
+        kind: Endpoints
+        apiVersion: v1
+        metadata:
+          name: docker-registry
+        subsets:
+          - addresses:
+              - IP: docker-registry-test.apps.example.com
+            ports:
+              - port: 443
+                name: docker-registry
+
+### Managing Authorization
+
+Cross-cluster authorization is a significant challenge. Automating or synchronizing cross-cluster authorization is ultimately desired. Until that can be achieved, manual processes may need to bridge the gap.
+
+**Deployments**
+
+Initially, a cluster administrator configures a service account with pull access to the appropriate registry environment so deployments "just work". This service account would have access to *all images* in the central registry irregardless of the environment. We are relying on each cluster registry service object to prevent pulling images from unintended environments. Additional engineering work would be required if restricting pulling images is required.
+
+**Builds**
+
+Build and push authorization is very important. Certain users or projects on certain clusters need access to certain registries. Initially, users could retrieve a "builder" service account token from the registry cluster for their namespace and create a *.dockercfg* build secret. This entails managing namespaces on two clusters and is not ideal.
+
+For organizations that maintain separate build clusters, the registry cluster could be part of this infrastructure cluster. This solves several authorization and namespace challenges.
```

Comment:
I'm skeptical of separate build clusters - it's not something we've said we'll support (we instead call them "dev clusters") explicitly so we can get away with it.